game/src/ecs/components/HealthComponent.java

```diff
@@ -0,0 +1,178 @@
+        this.currentHitPoints = maximalHitPoints;
+        this.onDeath = onDeath;
+        this.getHitAnimation = getHitAnimation;
+        this.dieAnimation = dieAnimation;
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Creates a HealthComponent with default values
+     *
+     * @param entity associated entity
+     */
+    public HealthComponent(@DSLContextMember(name = "entity") Entity entity) {
+        super(entity);
+        this.maximalHitPoints = 1;
+        this.currentHitPoints = 1;
+        this.onDeath = entity2 -> {};
+        this.getHitAnimation = new Animation(missingTexture, 100);
+        this.dieAnimation = new Animation(missingTexture, 100);
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Adds damage, which is accounted for by the system
+     *
+     * @param damage Damage that should be inflicted
+     */
+    public void getHit(Damage damage) {
+        damageToGet.add(damage);
+    }
+
+    /** Triggers the onDeath Function */
+    public void triggerOnDeath() {
+        onDeath.onDeath(entity);
+    }
+
+    /**
+     * @return List with all damage objects that still need to be accounted for
+     */
+    public List<Damage> getDamageList() {
+        return damageToGet;
+    }
+
+    /**
+     * Calculate the amount of damage of a certain type
+     *
+     * @param dt Type of damage object that still need to be accounted for
+     * @return Sum of all damage objects of type dt (default: 0)
+     */
+    public int getDamage(DamageType dt) {
+        return damageToGet.stream()
+                .filter(d -> d.damageType() == dt)
+                .mapToInt(Damage::damageAmount)
+                .sum();
+    }
+
+    /** Clear the damage list */
+    public void clearDamageList() {
+        damageToGet.clear();
+    }
+
+    /**
+     * Sets the current life points, capped at the value of the maximum hit-points
+     *
+     * @param amount new amount of current hit-points
+     */
+    public void setCurrentHitPoints(int amount) {
+        this.currentHitPoints = Math.min(maximalHitPoints, amount);
+    }
+
+    /**
+     * Sets the value of the Maximum Hit-points. If the new maximum hit-points are less than the
+     * current hit-points, the current points are set to the new maximum hit-points.
+     *
+     * @param amount new amount of maximal hit-points
+     */
+    public void setMaximalHitPoints(int amount) {
+        this.maximalHitPoints = amount;
+        currentHitPoints = Math.min(currentHitPoints, maximalHitPoints);
+    }
+
+    /**
+     * Set the animation to be played when the entity dies
+     *
+     * @param dieAnimation new dieAnimation
+     */
+    public void setDieAnimation(Animation dieAnimation) {
+        this.dieAnimation = dieAnimation;
+    }
+
+    /**
+     * Set the animation to be played when the entity is hit
+     *
+     * @param isHitAnimation new isHitAnimation
+     */
+    public void setGetHitAnimation(Animation isHitAnimation) {
```

Comment:
Diese Methode ist konzeptionell fragwürdig: Sollte die Animation nicht beim Erstellen der Component fest sein (und bleiben)? Gibt es einen Use-Case für das Ändern der Animation in der HealthComponent zu einem späteren Zeitpunkt? Im Code habe ich nur einen Test gefunden, sonst wird die Methode nicht verwendet.

Reply:
S.o